pkg/workload/workload_generator/ddl_generator.go

```diff
@@ -0,0 +1,368 @@
+	for _, p := range partsList {
+		up := strings.ToUpper(strings.TrimSpace(p))
+		if strings.HasPrefix(up, "CONSTRAINT") || strings.HasPrefix(up, "PRIMARY KEY") || strings.HasPrefix(up, "UNIQUE") || strings.HasPrefix(up, "FOREIGN KEY") || strings.HasPrefix(up, "CHECK") || strings.HasPrefix(up, "INDEX") {
+			tableConstraints = append(tableConstraints, p)
+		} else {
+			colDefs = append(colDefs, p)
+		}
+	}
+	// Define regex pattern to extract column properties from column definitions
+	colPattern := regexp.MustCompile(`(?i)^\s*("?[^"]+"|[\w-]+)"?\s+([^\s]+)(?:\s+(NOT\s+NULL|NULL))?(?:\s+DEFAULT\s+((?:\([^\)]*\)|[^\s,]+)))?(?:\s+PRIMARY\s+KEY)?(?:\s+UNIQUE)?(?:\s+REFERENCES\s+([\w\.]+)\s*\(\s*([\w]+)\s*\))?(?:\s+CHECK\s*\(\s*(.*?)\s*\))?`)
+	inlinePKCols := []string{}
+	// Process each column definition
+	for _, cd := range colDefs {
+		m := colPattern.FindStringSubmatch(cd)
+		if m == nil {
+			continue // Skip if pattern doesn't match
+		}
+		// Extract column properties from regex matches
+		name := m[1]             // Column name
+		columnType := m[2]       // Column type
+		nullSpec := m[3]         // NULL or NOT NULL specification
+		defaultVal := m[4]       // DEFAULT value
+		foreignKeyTable := m[5]  // Referenced table for foreign keys
+		foreignKeyColumn := m[6] // Referenced column for foreign keys
+
+		ts.ColumnOrder = append(ts.ColumnOrder, name)
+		// Extract CHECK constraint if present (requires special handling for nested parentheses)
+		inlineCheck := ""
+		checkIdx := regexp.MustCompile(`(?i)\bCHECK\s*\(`).FindStringIndex(cd)
+		if checkIdx != nil {
+			start := checkIdx[1]
+			depth := 1
+			i := start
+			for i < len(cd) && depth > 0 {
+				switch cd[i] {
+				case '(':
+					depth++
+				case ')':
+					depth--
+				}
+				i++
+			}
+			inlineCheck = strings.TrimSpace(cd[start : i-1])
+		}
+		// Determine some column properties
+		isNullable := nullSpec == "" || strings.ToUpper(nullSpec) == "NULL"
+		isUnique := regexp.MustCompile(`(?i)\bUNIQUE\b`).MatchString(cd)
+		isPK := regexp.MustCompile(`(?i)\bPRIMARY\s+KEY\b`).MatchString(cd)
+
+		// Handle inline PRIMARY KEY constraints
+		if isPK {
+			inlinePKCols = append(inlinePKCols, strings.Trim(name, `"`))
+			isNullable = false // PRIMARY KEY columns cannot be NULL
+			isUnique = true    // PRIMARY KEY columns are implicitly UNIQUE
+		}
+		// Create and populate the Column object
+		col := &Column{
+			Name:         strings.Trim(name, `"`),
+			ColType:      columnType,
+			IsNullable:   isNullable,
+			IsPrimaryKey: isPK,
+			Default:      strings.TrimSpace(defaultVal),
+			IsUnique:     isUnique,
+		}
+		// Add foreign key information if present
+		if foreignKeyTable != "" {
+			col.FKTable = foreignKeyTable
+			col.FKColumn = foreignKeyColumn
+		}
+		// Add CHECK constraint if present
+		if inlineCheck != "" {
+			col.InlineCheck = inlineCheck
+			ts.CheckConstraints = append(ts.CheckConstraints, inlineCheck)
+		}
+		ts.AddColumn(col)
```

Comment:
instead of this, lets make a function as "AddColumns" which takes colDefs as input and does the loop in that function itself?